grpc/creds/creds.go

```diff
@@ -13,28 +12,41 @@ import (
 // transportCredentials is a grpc/credentials.TransportCredentials which supports
 // connecting to, and verifying multiple DNS names
 type transportCredentials struct {
-	roots   *x509.CertPool
-	clients []tls.Certificate
+	clientConfig *tls.Config
```

Comment:
@jsha I agree it would be clearer split into one type for a client and one for the server. I'll make that change.

Switching the `rootCA` and `clientCerts` to a `tls.Config` was to keep the `New()` function arguments clean, but at the time I didn't think about the complication of not being able to mutate the servername and having to pick/choose fields. I probably should have revisited the decision at that point instead of continuing :-) With your suggestion to split the `transportCredenetials` type I think it makes sense to restore the original parameters for the client version.

Thanks!